Idea is to have two nonintersecting SliTaz packages:
gucharmap (for GTK+2) and gucharmap3 (for GTK+3)
Author: Aleksej Bobylev <al.bobylev@gmail.com>

Plus: https://gitlab.gnome.org/GNOME/gucharmap/commit/4ad97abc893ded93296f7c157631369fbddcf90a

--- a/configure.ac
+++ b/configure.ac
@@ -36,7 +36,7 @@
 
 AC_PREREQ([2.56])
 
-AC_INIT([GNOME Character Map],[gucharmap_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=gucharmap],[gucharmap])
+AC_INIT([GNOME Character Map],[gucharmap_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=gucharmap],[gucharmap3])
 
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([gucharmap/gucharmap.h.in])
@@ -48,7 +48,7 @@
 
 # checks for progs
 
-GNOME_DEBUG_CHECK
+#GNOME_DEBUG_CHECK
 
 AC_PROG_CC
 AC_USE_SYSTEM_EXTENSIONS
@@ -120,6 +120,13 @@
   freetype2
 ])
 
+sv_LIBS="$LIBS"
+LIBS=
+AC_SEARCH_LIBS([dlsym],[dl])
+LIBDL="$LIBS"
+LIBS="$sv_LIBS"
+AC_SUBST([LIBDL])
+
 GLIB_GENMARSHAL="$($PKG_CONFIG --variable=glib_genmarshal glib-2.0)"
 AC_SUBST([GLIB_GENMARSHAL])
 GLIB_MKENUMS="$($PKG_CONFIG --variable=glib_mkenums glib-2.0)"
@@ -225,7 +232,7 @@
 # i18n
 # ****
 
-GETTEXT_PACKAGE=gucharmap
+GETTEXT_PACKAGE=gucharmap3
 AC_SUBST([GETTEXT_PACKAGE])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[The gucharmap gettext domain])
 AM_GLIB_GNU_GETTEXT
@@ -322,7 +329,7 @@
 
 AC_CONFIG_FILES([
 Makefile
-gucharmap.desktop.in
+gucharmap3.desktop.in
 gucharmap.nsi
 gucharmap.spec
 data/Makefile
--- /dev/null
+++ b/gucharmap3.desktop.in.in
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+_Name=Character Map
+_Comment=Insert special characters into documents
+StartupNotify=true
+Exec=gucharmap3
+Icon=accessories-character-map
+Categories=GNOME;GTK;Utility;
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,7 +14,7 @@
 distcleancheck_listfiles = find . -type -f print | grep -v 'omf\.out'
 
 desktopdir = $(datadir)/applications
-desktop_in_files = gucharmap.desktop.in.in
+desktop_in_files = gucharmap3.desktop.in.in
 desktop_DATA 	 = $(desktop_in_files:.desktop.in.in=.desktop)
 
 pkgconfigdir = $(libdir)/pkgconfig
--- a/gucharmap/Makefile.am
+++ b/gucharmap/Makefile.am
@@ -156,6 +156,7 @@
 gucharmap_LDADD = \
 	libgucharmap@GUCHARMAP_LIBRARY_SUFFIX_U@.la	\
 	$(GTK_LIBS)	\
+	$(LIBDL) \
 	$(INTL_LIBS)
 
 dist_noinst_DATA = gucharmap-menus.ui
